go/pkg/cas/upload.go

```diff
@@ -21,8 +23,16 @@ import (
 	"github.com/bazelbuild/remote-apis-sdks/go/pkg/digest"
 	"github.com/bazelbuild/remote-apis-sdks/go/pkg/retry"
 	repb "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2"
+	bspb "google.golang.org/genproto/googleapis/bytestream"
 )
 
+var zstdEncoders = sync.Pool{
+	New: func() interface{} {
+		enc, _ := zstd.NewWriter(nil)
+		return enc
+	},
```

Comment:
If you're not going to use zstdpool here I would recommend leaving a comment about not forgetting to call Close on encoders, it's a subtle gotcha waiting to happen.

Reply:
done

Reply:
While I agree that consistency would be nice, my main concern, which I didn't mention anywhere, is the use of finalizers in zstd-pool. They feel like a bit overkill. Dave Cheney recommends against them and provides some history around finalizers here: https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_avoid_finalisation
His main point is in the end: 

> Do not write programs who’s correctness depends on finalisation, instead associate the lifetime of a resource to the lifetime of a goroutine.

which is what this code is doing (except, this is not about correctness).

Thoughts?

Reply:
I agree in principle that using finalizers isn't great, but fixing that properly would require significant changes to the zstd library because it breaks this other one of Dave Cheney's tips:
https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_leave_concurrency_to_the_caller

Reply:
I agree that it would be hard to fix it at the zstd package level.

Reply:
sgtm, let's punt